spec/lib/jit_preloader/preloader_spec.rb

```diff
@@ -165,6 +165,33 @@
           expect(Contact.jit_preload.map(&:contact_owner)).to eq [nil, ContactOwner.first, Address.first]
         end
       end
+
+      context "when a record has a polymorphic association type is nil" do
+        before do
+          contact1.update!(contact_owner_type: nil, contact_owner_id: nil)
+        end
+
+        it "successfully load the rest of association values" do
+          contacts = Contact.jit_preload.to_a
+          expect(contacts.first.contact_owner).to eq(nil)
```

Comment:
So the query run here - is it a false alarm or is actually running a query?  Is there  anything we can do to mark the association as loaded isntead of running a query?

Reply:
It's running the preload but ActiveRecord marks the preload as `loaded unsuccessfully` since it can not successfully interpret the return type (which is nil in this case).

My primitive idea is to add another check on here https://github.com/clio/jit_preloader/blob/b1a60fde8d4cd3054735324bc3349d50395b1377/lib/jit_preloader/active_record/associations/singular_association.rb#L5
that just return if we found a record match the following conditions
1. the record target association to load is a polymorphic one, this should easily check on the reflection itself.
2. the record association type is nil.

But that will have to come as another PR.

Reply:
no, cause running the `contacts.first.contact_owner` is the first time it preload the `contact_owner` association which should execute the query. The following access for the same association is the part it should not execute another query to access the data.

Reply:
If you repeat this line in the `expect { ... }.not_to make_dsatabase_queries` section what happens?  That seems like that would prove the issue is a false positive; at the moment, we've asserted that for two of the records we don't have a problem with our polymorphic code, why not the first?

Reply:
This is my local outcome if we wrap the first one in expect too
![image](https://github.com/clio/jit_preloader/assets/139145049/e3fac842-70f8-428e-b304-4695f60d2d97)

with code
```
it "successfully load the rest of association values" do
            contacts = Contact.jit_preload.to_a
            expect do
              contacts.first.contact_owner
            end.not_to make_database_queries
  
            expect do
              contacts.second.contact_owner
              contacts.third.contact_owner
            end.not_to make_database_queries
  
            expect(contacts.second.contact_owner).to eq(ContactOwner.first)
            expect(contacts.third.contact_owner).to eq(Address.first)
end
```

Reply:
sure, I can do that.